libs/sdk-core/src/lib/db/kb/kb.ts

```diff
@@ -115,16 +115,64 @@ export class KnowledgeBox implements IKnowledgeBox {
     ],
   ): Observable<Resource> {
     const params = [...show.map((s) => `show=${s}`), ...extracted.map((e) => `extracted=${e}`)];
-    const path = !uuid ? `${this.path}/slug/${slug}` : `${this.path}/resource/${uuid}`;
     return this.nuclia.rest
-      .get<IResource>(`${path}?${params.join('&')}`)
+      .get<IResource>(`${this._getPath(uuid, slug)}?${params.join('&')}`)
       .pipe(map((res) => new Resource(this.nuclia, this.id, res)));
   }
 
+  private _getPath(uuid?: string, slug?: string): string {
+    return !uuid ? `${this.path}/slug/${slug}` : `${this.path}/resource/${uuid}`;
+  }
+
   getResourceFromData(data: IResource): Resource {
     return new Resource(this.nuclia, this.id, data);
   }
 
+  getResourceField(
```

Comment:
it should belong to the `ReadableResource` class (and actually there is already a `getField` method, it just does not supports the `show` and `extracted` options but they can be implemented)

Reply:
putting this in `ReadableResource` means we need to have a resource object to be able to call this method though, which is not always the case

Reply:
There is a `getFields` method in ReadableResource, returning a list of fields, not a `getField` method returning a specific field.